### DISCUSSION TITLE
Search endpoint needs to return title and artist

### DISCUSSION THREAD

**Discussion by gigbuilder** [tidal-music/discussions#94] [Q&A]
The search utility is pretty lame (so is most of the API at this moment) and I cannot believe the search results (I'm doing by track) include the various songs found but NO ARTIST tag?  How does one ensure the song they are searching is correct without knowing the artist or even album that the song associates?  If there's a way to do this I'd sure like to know and if there isn't I'd sure like a fix.

**Comment by stefanmoro** [comment]
Hey @gigbuilder,

The data model exposed on TIDAL API links artists from a track via the `artists` relationship, you can check out the `tracks` definition [e.g. here](https://developer.tidal.com/apiref?spec=catalogue-v2&ref=get-track-v2).

Today, the TIDAL API supports one level of relationship resolvement, so for your specific use case I'd recommend the follwowing:
1. `/searchresults/yoursearchquery?include=tracks`
2. `/tracks?filter[id]=id1,id2...,idx&include=artists,albums`

We are discussing/evaluating support for nested inclusion, but it isn't currently supported.

**Comment by davddavd** [comment]
Unless I am mistaken the search results are returning the same tracks over and over in each "next" link.  The number of pages seems to be capped at 15 or less depending on the search terms.  For example:

Searching The Smiths using type tracks
page: 1
/searchresults/The+Smiths/relationships/tracks?include=tracks&countryCode=US&page%5Bcursor%5D=3nI1Esi
tracks on page
7932939,7861997,7915595,7932950,7932936,7932957,7932933,7932946,7932951,7915655,3268387,7915605,7915598,7915634,68737324,68737343,68737342,7915639,375506094,52885535
page: 2
/searchresults/The+Smiths/relationships/tracks?include=tracks&countryCode=US&page%5Bcursor%5D=3nI1F0y
tracks on page
7932939,7861997,7915595,7932950,7932936,7932957,7932933,7932946,7932951,7915655,3268387,7915605,7915598,7915634,68737324,68737343,68737342,7915639,375506094,52885535
...snipped ...
page: 14
/searchresults/The+Smiths/relationships/tracks?include=tracks&countryCode=US&page%5Bcursor%5D=FfWP7SRU
tracks on page
7932939,7861997,7915595,7932950,7932936,7932957,7932933,7932946,7932951,7915655,3268387,7915605,7915598,7915634,68737324,68737343,68737342,7915639,375506094,52885535
page: 15
/searchresults/The+Smiths/relationships/tracks?include=tracks&countryCode=US&page%5Bcursor%5D=FfWP7ixU
tracks on page
7932939,7861997,7915595,7932950,7932936,7932957,7932933,7932946,7932951,7915655,3268387,7915605,7915598,7915634,68737324,68737343,68737342,7915639,375506094,52885535

I use almost an identical procedure for playlists and the pages are correct such that I get a list of tracks from the next page and then query multiple tracks with filter[ids] to get the remaining information page by page.

Am I mistaken about the search results returning the same tracks in error?

**Reply by stefanmoro** [reply]
Hey @davddavd ,

You're right, there was a bug in our pagination. Fixed now, thank you for pointing this out!

**Reply by davddavd** [reply]
The pagination bug is still present.  Sorry but I posted "Thanks for fixing that" prematurely.

**Reply by stefanmoro** [reply]
Hi again,

We just tested ourselves and it seems to work fine for us. Can you please post the exact requests you are making? Also, please note that as of now, including playlists from search results require an access token acquired using authorization code flow (we aim to change that in the future, and allow also client credentials access tokens for public playlists.)

Thank you

**Reply by davddavd** [reply]
It is working now.

**Reply by davddavd** [reply]
The items are included in the next page links correctly now but unfortunately "next" links are provided in error past the 15 page/300 item threshold that produce a 500 Internal Server Error on the sixteenth page.  Thanks for addressing these issues so quickly with actual bug fixes.

**Reply by kristianpettersen** [reply]
Hi @davddavd 

This is fixed now, the next link should no longer be returned when you reach the pagination limit.

**Comment by davddavd** [comment]
Thanks for fixing that.  Also, I get internal server errors for search version 1 queries every once and a while that are dependent on the content of the query and not the syntax.  For example the query 
https://openapi.tidal.com/search?query=rolling+stones&offset=0&limit=88&countryCode=US&popularity=WORLDWIDE
produces an error like:
{"errors":[{"category":"API_ERROR","code":"INTERNAL_SERVER_ERROR","detail":"The service encountered an error"}]}
but the query
https://openapi.tidal.com/search?query=rolling+stones&offset=0&limit=87&countryCode=US&popularity=WORLDWIDE 
succeeds.  The exact limit number changes from time to time for this type of error so I use a limitless and pageless query like
str=getURI(new URI("https://openapi.tidal.com/search?query=" + query + "&countryCode=US")); 

A query using a too high limit value such as 101 returns
{"errors":[{"category":"INVALID_REQUEST_ERROR","code":"VALUE_TOO_HIGH","detail":"must be less than or equal to 100","field":"limit"}]}

**Reply by stefanmoro** [reply]
Hi @davddavd,

We will sunset the entire "V1" API relatively soon (we already removed it from our [API reference](https://developer.tidal.com/reference/web-api) some time ago), all future development and fixes will be made on "V2". 

Thanks
